clang/lib/Parse/Parser.cpp

```diff
@@ -1196,9 +1196,20 @@ void Parser::ProcessPluginSyntax(ParsingDeclarator &D) {
   llvm::raw_string_ostream ReplacementOS(Replacement);
 
   ReplacementOS << "\n{\n";
-  SHI->second->GetReplacement(PP, D, Toks, ReplacementOS);
   ReplacementOS << "\n}\n";
+  //  provide token stream to the plugin and get back replacement text
+  //  but place them just outside the function body been replaced
+  SHI->second->GetReplacement(PP, D, Toks, ReplacementOS);
   ReplacementOS.flush();
+  
+  //Now we change the identifier name in the declarator to forget
```

Comment:
Same here (plus there should be a space between the `//` and the text.

Reply:
Okay, done.